api/v1alpha1/shared_types.go

```diff
@@ -579,6 +579,22 @@ type ClusterSettings struct {
 	//
 	// +optional
 	HTTP2 *HTTP2Settings `json:"http2,omitempty"`
+
```

Comment:
my preference would be to add a `PanicThreshold unit32` field inside `HealthCheck`
https://github.com/envoyproxy/gateway/blob/0e52d06721737bdcbc9a1e8ffe969a4acd697a25/api/v1alpha1/healthcheck_types.go#L17